clarity/src/vm/ast/parser_v2/lexer/mod.rs

```diff
@@ -0,0 +1,2207 @@
+pub mod error;
+pub mod token;
+
+use crate::vm::{
+    diagnostic::{DiagnosableError, Level},
+    representations::Span,
+};
+use std::{char, str::Chars};
+use token::{PlacedToken, Token};
+
+use crate::vm::ast::parser_v2::lexer::error::LexerError;
+
+use self::error::PlacedError;
+
+pub type LexResult<T> = Result<T, LexerError>;
+
+const EOF: char = std::char::REPLACEMENT_CHARACTER;
+
+pub struct Lexer<'a> {
+    input: Chars<'a>,
+    next: char,
+    offset: usize,
+    pub line: usize,
+    pub column: usize,
+    pub last_line: usize,
+    pub last_column: usize,
+    // While lexing, collect diagnostics and continue on (when possible)
+    pub diagnostics: Vec<PlacedError>,
+    pub success: bool,
+    // Used to report an error for line-endings only on the first instance
+    line_endings: bool,
+    // `fail_fast` mode indicates that the lexer should not report warnings
+    // and should exit on the first error. This is useful for lexing in the
+    // context of a stacks-node, while normal mode is useful for developers.
+    fail_fast: bool,
+}
+
+fn is_separator(ch: char) -> bool {
+    match ch {
+        '(' | ')' | '{' | '}' | ',' | ':' | '.' | ';' | EOF => true,
+        _ => ch.is_ascii_whitespace(),
+    }
+}
+
+fn is_string_terminator(ch: char) -> bool {
+    match ch {
+        '"' | '\n' | EOF => true,
```

Comment:
Does this mean that the following contract will parse?

```clarity
"hello world
```

I think you can have "bare" contracts that don't have any lists, but just raw literals.  I just want to make sure that calling `EOF` a string terminator doesn't lead to silliness like this.

Reply:
No, this is just used to read through the "end" of a string when an error is encountered. If a `"` is not found before the end of the line or the end of the file, then the string is considered closed. This is only reached when not in "fail fast" mode, when attempting to continue parsing past an error.